lighthouse-core/gather/computed/computed-artifact.js

```diff
@@ -5,13 +5,15 @@
  */
 'use strict';
 
+const CustomMap = require('../../lib/custom-map');
```

Comment:
this impl doesn't support `new CustomMap([[x, 1], [y, 2]])`. 

given that maybe call it `CacheMap`?    really anything to give it some distance appearing as a true subclass of `Map`

Reply:
ah yeah, good call I'll add support :)

Reply:
hmmm. I don't think we have a real usecase for supporting that constructor API. can we keep it out of scope?

Reply:
alright alright, cache map it is